subsys/bluetooth/host/l2cap.c

```diff
@@ -2296,7 +2296,10 @@ static void l2cap_chan_send_credits(struct bt_l2cap_le_chan *chan,
 	struct bt_l2cap_le_credits *ev;
 	struct net_buf *buf;
 
-	__ASSERT_NO_MSG(bt_l2cap_chan_get_state(&chan->chan) == BT_L2CAP_CONNECTED);
+	if (bt_l2cap_chan_get_state(&chan->chan) != BT_L2CAP_CONNECTED) {
+		LOG_ERR("L2CAP channel not connected: %d", bt_l2cap_chan_get_state(&chan->chan));
+		return;
+	}
```

Comment:
I think this is just masking the underlying issue. The assert was likely put there for a reason, so if we hit it, then chances are the the caller of `l2cap_chan_send_credits` is doing something wrong. 

Can you elaborate on when this happens? Preferably with some logs with at least L2CAP debug logs enabled?

Reply:
I agree this may be masking some underlying problem.

I think the assertion in question is inappropriate, since `l2cap_chan_send_credits` is invoked from `bt_l2cap_chan_recv_complete`, a public API, so potentially on a preemptible thread racing with a disconnection.

The new if-check is also pointless for the same reason.